src/tools/BaseTool.ts

```diff
@@ -35,19 +39,51 @@ export default abstract class BaseTool {
     }
     const convertedPoint = point.matrixTransform(invertedCTMMatrix);
 
-    return [
-      BaseTool.roundCoordinateToGrid(convertedPoint.x),
-      BaseTool.roundCoordinateToGrid(convertedPoint.y),
-    ];
+    return [convertedPoint.x, convertedPoint.y];
+  }
+
+  /**
+   * Convert clientX/Y to the X/Y coordinates on the SVG rectangle.
+   **/
+  static convertClientCoordinatesRounded(event: MouseEvent): [number, number] {
+    const point: [number, number] = BaseTool.convertClientCoordinates(event);
+    return BaseTool.roundCoordinateToGrid(point);
   }
 
-  /* eslint-disable @typescript-eslint/no-unused-vars,
-    @typescript-eslint/no-empty-function */
-  onClick(event: MouseEvent): void {}
+  /**
+   * returns index of dot at mouse event, or -1 if nothing found
+   **/
+  static findDotAtEvent(event: MouseEvent): number {
+    const [x, y] = BaseTool.convertClientCoordinatesRounded(event);
+    const stuntSheetDots: StuntSheetDot[] =
+      GlobalStore.getters.getSelectedStuntSheet.stuntSheetDots;
+    return stuntSheetDots.findIndex((dot: StuntSheetDot): boolean => {
+      return x === dot.x && y === dot.y;
+    });
+  }
+
+  /**
+   * Convert clientX/Y to the X/Y coordinates on the SVG rectangle.
+   **/
+  static updateInvertedCTMMatrix(): void {
+    /**
+     * Calculate inverted CTM matrix that is used to convert ClientX/Y to
+     * X/Y of the SVG
+     */
+    const wrapper = document.getElementsByClassName(
+      "grapher--wrapper"
+    )[0] as SVGGElement;
+    const ctm = wrapper.getScreenCTM();
+    if (!ctm) {
+      throw new Error("Unable to retrieve wrapper CTM");
+    }
+    const invertedMatrix = ctm.inverse();
+    GlobalStore.state.invertedCTMMatrix = invertedMatrix;
```

Comment:
Can you change this to a mutation on the vuex store? We should try to avoid directly modifying the state outside of mutations.

Reply:
yep!

Reply:
yep!